### DISCUSSION TITLE
Logs size

### DISCUSSION THREAD

**Discussion by vijexa** [scalameta/metals#2309] [Q&A] [answered]
I was shocked to find out that my simple project almost without dependencies that just contains solutions for AoC tasks was taking around 500MB of space. Upon closer inspection turns out that there are almost half a gig of logs... I don't know if this is a bug or maybe I'm doing something wrong, so I'm not posting an issue yet, but this can't be right. Should I open an issue?

```shell
vijexa@DESKTOP-ETMRL27:~/advent-of-code-2020/advent-of-code-2020/.metals$ du -ahd 1
644K    ./metals.h2.db
488M    ./metals.log
804K    ./readonly
489M    .
```

**Comment by ckipp01** [comment, marked as answer]
Good question @vijexa. Just to give a comparison my `metals.log` file in the metals codebase itself is
```
6.0M    ./metals.log
```
So I really don't think it should be as large as you're showing. I want to guess that you may have a bunch of exceptions or something being thrown in there, or something else causing an excessive amount of output. Are you able to take a peak inside and browse around? Report back anything that looks out of places like exceptions etc.

**Reply by vijexa** [reply]
There are a few exceptions here and there, but what I found was more surprising than that. For some reason, all my programs output to stdout is logged there. As I said, this is an AoC project, so there is often a ton of data that needs to be parsed or something. So, a lot of printing to check if stuff works correctly. And I guess I managed to print 500MB of text in two weeks. This is hilarious. Is there a good reason why Metals logs program output?

**Reply by tgodzik** [reply]
I think it's due to the fact that Bloop reports it when running, I think we should not write it to log, but I can't remember why it was done in the first place. This should become an issue really.

**Reply by ckipp01** [reply]
Huh, that's actually a great question. I actually have no idea. I'll need to dig in a bit to see. I'm pretty unfamiliar with a lot of our DAP related stuff, so one of the other maintainers may have a better idea.

**Reply by ckipp01** [reply]
whoop, ha, I didn't see you already answered @tgodzik. I'll go ahead and convert this to an issue.

**Reply by vijexa** [reply]
Thanks 🙂
